src/ad/azext_ad/manual/custom.py

```diff
@@ -0,0 +1,195 @@
+
+    if any([settings, resource_forest]):
+        resource_forest_settings = domain_service.resource_forest_settings
+        new_domain_service['resource_forest_settings'] = {}
+        new_domain_service['resource_forest_settings']['settings'] = \
+            _get_domain_service_property_value(resource_forest_settings, 'settings', None) if settings is None else _get_resource_forest_settings(settings)
+        new_domain_service['resource_forest_settings']['resource_forest'] = \
+            _get_domain_service_property_value(resource_forest_settings, 'resource_forest', None) if resource_forest is None else resource_forest
+
+    if any([ldaps, pfx_certificate, pfx_certificate_password, external_access]):
+        ldaps_settings = domain_service.ldaps_settings
+        pfx_certificate = _get_pfx_certificate_content(pfx_certificate)
+        new_domain_service['ldaps_settings'] = {}
+        new_domain_service['ldaps_settings']['ldaps'] = \
+            _get_domain_service_property_value(ldaps_settings, 'ldaps', 'Disabled') if ldaps is None else ldaps
+        new_domain_service['ldaps_settings']['pfx_certificate'] = \
+            _get_domain_service_property_value(ldaps_settings, 'pfx_certificate', None) if pfx_certificate is None else pfx_certificate
+        new_domain_service['ldaps_settings']['pfx_certificate_password'] = \
+            _get_domain_service_property_value(ldaps_settings, 'pfx_certificate_password', None) if pfx_certificate_password is None else pfx_certificate_password
+        new_domain_service['ldaps_settings']['external_access'] = \
+            _get_domain_service_property_value(ldaps_settings, 'external_access', 'Disabled') if external_access is None else external_access
+
+    return sdk_no_wait(no_wait,
+                       client.begin_update,
+                       resource_group_name=resource_group_name,
+                       domain_service_name=domain_service_name,
+                       domain_service=new_domain_service)
+
+
+def _get_domain_service_property_value(old_property, sub_property_name, default_value):
+    if old_property is not None:
+        return getattr(old_property, sub_property_name, default_value)
+    return default_value
+
+
+def _get_resource_forest_settings(settings):
+    if not settings:
+        return settings
+    import os
+    from azure.cli.core.util import get_file_json, shell_safe_json_parse
+    if os.path.exists(settings):
+        return get_file_json(settings)
+    else:
+        return shell_safe_json_parse(settings)
+
+
+def _get_pfx_certificate_content(pfx_certificate):
+    if not pfx_certificate:
+        return pfx_certificate
+    import os
+    from azure.cli.core.util import read_file_content
+    if os.path.exists(pfx_certificate):
+        return read_file_content(pfx_certificate)
+    return pfx_certificate
```

Comment:
Is it possible to move it to validation part?

Reply:
Moved the last two to validators, the first one is to read existing property value after `client.get` so I kept it within `custom.py`.